src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -396,10 +396,23 @@ private List<Feedback> saveFeedbackWithHibernateWorkaround(@NotNull Result resul
         feedbackList.forEach(feedback -> {
             // cut association to parent object
             feedback.setResult(null);
+            LongFeedbackText longFeedback = null;
+            // search for long feedback parent feedback has and cut the association between them.
+            if (feedback.getId() != null) {
```

Comment:
Possible optimisation to avoid unnecessary work here and in the similar `if` below:
```suggestion
            if (feedback.getId() != null && feedback.hasLongFeedbackText()) {
```
That should be stored as direct attribute in the feedback and still be `true` even if the connected long feedback has not been loaded.